lib/game.rb

```diff
@@ -31,123 +35,102 @@ def self.count_of_games_by_season
   def self.lowest_scoring_visitor_team_id
-    avg_away_goals = average_goals(goals_per(:away_team_id, :away_goals), games_per(:away_team_id))
-    away_team_ids = @@all.map { |game| game.away_team_id }.uniq
-    # create hash of {team_ids => average goals}
-    away_ids_n_goals = Hash[away_team_ids.zip(avg_away_goals)]
-    away_ids_n_goals.min_by{ |team_id, away_goals| away_goals}.first
+    average_goals_by(:away_team).min_by{ |team_id, away_goals| away_goals}.first
   end
-#deliverable (needs to access teams.csv via stat_tracker)
+#deliverable
   def self.lowest_scoring_home_team_id
-    avg_home_goals = average_goals(goals_per(:home_team_id, :home_goals), games_per(:home_team_id))
-    home_team_ids = @@all.map { |game| game.home_team_id }.uniq
-    # create hash of {team_ids => average goals}
-    home_ids_n_goals = Hash[home_team_ids.zip(avg_home_goals)]
-    home_ids_n_goals.min_by{ |team_id, home_goals| home_goals}.first
+    average_goals_by(:home_team).min_by{ |team_id, away_goals| away_goals}.first
   end
 
-  def self.games_by_season(team_id)
-
-    #accumulator hash
-    games_by_season = Hash.new { |hash, key| hash[key] = 0 }
-    @@all.each do |game|
-      #returns true if a team was in a given game
-      team_played_in_game = game.away_team_id == team_id || game.home_team_id == team_id
-      games_by_season[game.season] += 1 if team_played_in_game
+  def self.games_played_by(team_id)
+    #return all games that team played in
+    all.find_all do |game|
+      game.away_team_id == team_id || game.home_team_id == team_id
     end
-    games_by_season
   end
 
-  def self.wins_by_season(team_id)
-
-    season_wins = Hash.new { |hash, key| hash[key] = 0 }
-    @@all.each do |game|
-      #add 1 to season wins 
-      season_wins[game.season] += 1 if game.win?(team_id)
-      #returns true if a team was in a given game
-      team_played_in_game = game.away_team_id == team_id || game.home_team_id == team_id
-      #this below will populate a 0 if a team had no wins that season
-      season_wins[game.season] if team_played_in_game
-    end
-    season_wins
+  def self.games_and_wins_by_season(team_id)
+      #{ season => {:wins => x, :games_played => y}}
+    hash_of_hashes(games_played_by(team_id), :season, :wins, :games_played, :win?, 1, team_id)
   end
 
-  def self.percent_by_season(team_id)
-    #this each_with_index can perform mult/div on arrays of the same size
-    # [1,2,3] * [11,12,13] = [11,24,39]
-    num_games = games_by_season(team_id).values
-    percent_per_season = wins_by_season(team_id).values.each_with_index.map do |wins, index|
-      ((wins.to_f / num_games[index].to_f)*100).to_i
-    end
-
-    Hash[games_by_season(team_id).keys.zip(percent_per_season)]
+  def self.win_percent_by_season(team_id)
+    # :wins / :games_played * 100
+    win_percent_by_season = divide_hash_values(:wins, :games_played, games_and_wins_by_season(team_id))
+    win_percent_by_season.transform_values { |v| (v * 100).to_i}
   end
```

Comment:
This is an example of how to get a percentage with the `divide_hash_values` method. Since its general division, you have to multiply by 100 at the end